File: trunk/GPG.Multiplayer.Statistics/GPG/Multiplayer/Statistics/AwardCategory.vb
Imports GPG.DataAccess
Imports GPG.Multiplayer.Quazal
Imports System
Imports System.Collections.Generic

Namespace GPG.Multiplayer.Statistics
    Public Class AwardCategory
        Inherits MappedObject
        ' Methods
        Public Sub New(ByVal record As DataRecord)
            MyBase.New(record)
            Me.mSortOrder = -1
        End Sub

        Public Shared Sub ClearCachedData()
            AwardCategory.mAllAwardCategories = Nothing
        End Sub


        ' Properties
        Public Shared ReadOnly Property AllAwardCategories As Dictionary(Of Integer, AwardCategory)
            Get
                If (AwardCategory.mAllAwardCategories Is Nothing) Then
                    AwardCategory.mAllAwardCategories = New Dictionary(Of Integer, AwardCategory)
                    Dim category As AwardCategory
                    For Each category In New QuazalQuery("GetAllAwardCategories", New Object(0  - 1) {}).GetObjects(Of AwardCategory)
                        AwardCategory.mAllAwardCategories.Add(category.ID, category)
                    Next
                End If
                Return AwardCategory.mAllAwardCategories
            End Get
        End Property

        Public ReadOnly Property ID As Integer
            Get
                Return Me.mID
            End Get
        End Property

        Public ReadOnly Property Name As String
            Get
                Return Me.mName
            End Get
        End Property

        Public ReadOnly Property SortOrder As Integer
            Get
                Return Me.mSortOrder
            End Get
        End Property


        ' Fields
        Public Shared mAllAwardCategories As Dictionary(Of Integer, AwardCategory)
        <FieldMap("award_category_id")> _
        Private mID As Integer
        <FieldMap("name")> _
        Private mName As String
        <FieldMap("sort_order")> _
        Private mSortOrder As Integer
    End Class
End Namespace

File: GPG.Multiplayer.Statistics/GPG/Multiplayer/Statistics/AwardCategory.cs
namespace GPG.Multiplayer.Statistics
{
    using GPG.DataAccess;
    using GPG.Multiplayer.Quazal;
    using System;
    using System.Collections.Generic;

    public class AwardCategory : MappedObject
    {
        public static Dictionary<int, AwardCategory> mAllAwardCategories;
        [FieldMap("award_category_id")]
        private int mID;
        [FieldMap("name")]
        private string mName;
        [FieldMap("sort_order")]
        private int mSortOrder = -1;

        public AwardCategory(DataRecord record) : base(record)
        {
        }

        public static void ClearCachedData()
        {
            mAllAwardCategories = null;
        }

        public static Dictionary<int, AwardCategory> AllAwardCategories
        {
            get
            {
                if (mAllAwardCategories == null)
                {
                    mAllAwardCategories = new Dictionary<int, AwardCategory>();
                    foreach (AwardCategory category in new QuazalQuery("GetAllAwardCategories", new object[0]).GetObjects<AwardCategory>())
                    {
                        mAllAwardCategories.Add(category.ID, category);
                    }
                }
                return mAllAwardCategories;
            }
        }

        public int ID
        {
            get
            {
                return this.mID;
            }
        }

        public string Name
        {
            get
            {
                return this.mName;
            }
        }

        public int SortOrder
        {
            get
            {
                return this.mSortOrder;
            }
        }
    }
}

